sensor_test.py

```diff
@@ -55,6 +56,38 @@ def test_set_sensor_name(get_sensor_info, set_sensor_name):
     ), f"Sensor name was not updated, expected name:{new_sensor_name}"
 
 
+def test_set_empty_sensor_name(get_sensor_info, set_sensor_name):
+    """
+    1. Get original sensor name.
+    2. Set sensor name to an empty string.
+    3. Validate that sensor responds with an error.
+    4. Get current sensor name.
+    5. Validate that sensor name didn't change.
+    """
+
+    log.info("Get original sensor name")
+    sensor_info = get_sensor_info()
+    original_sensor_name = sensor_info.name
+
+    log.info("Set sensor name to an empty string")
+    set_empty_sensor_name = set_sensor_name("")
+
+    log.info("Validate that sensor responds with an error")
+    if "error" in set_empty_sensor_name:

```

Comment:
The error is present in this response but not in the form you're looking for. The changes you made to `make_valid_request()` return the contents of the `"error"` key if it's present in the response from the sensor. So, here instead of checking for `"error"` you need to be checking for `"code"` or `"message"`.

That would be useful if you're interested in exactly the type of error the sensor returned. Let's say for simplicity's sake that presently we only care whether we get an error at all, regardless of its type. We can do that without making changes to `make_valid_request()`.

We're calling `set_sensor_name()` to set sensor name to an invalid value. If we do this:
```
response = set_sensor_name("valid_name")
```
The value in `"response"` would be a dict containing sensor info.

If we do this:
```
response = set_sensor_name("")
```
The value in response would be an empty dict. This is how we know the sensor returned an error.